src/legacy/core_plugins/embeddable_api/public/types.ts

```diff
@@ -0,0 +1,83 @@
+/*
```

Comment:
I think this is somewhat debateable.  IEmbeddable for instance pulls in Adapters from `ui/inspector`.  So I *think* that if someone just wants to use the `Query` type, it will end up pulling in the Adapters class from `ui/inspectors`.  So you could throw them all in one file, but you can end up importing more unnecessary static code that way (if my assumptions are correct).

Reply:
> I think this is somewhat debateable.

It is. It's just preference thing.

> ... but you can end up importing more unnecessary static code that way.

I meant to put only TypeScript types in `types.ts`, no code.

Reply:
ahhh, right, it'll get compiled away